### DISCUSSION TITLE
Alacritty terminfo files not installed by default

### DISCUSSION THREAD

**Discussion by sluukkonen** [Homebrew/discussions#1306] [Casks] [answered]
The [alacritty](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/alacritty.rb) cask includes terminfo entries in `/Applications/Alacritty.app/Contents/Resources` folder, but they aren't copied to a terminfo database during installation.

Question: Is this by design? For now, I've solved the problem by copying the terminfo entries at `/Applications/Alacritty.app/Contents/Resources/61` to the `~/.terminfo` directory, but it doesn't seem ideal that every user has to do this manually.

The regular terminfo database at `/usr/share/terminfo` seems to be read-only in Big Sur, so using it is probably out of the question.

**Comment by carlocab** [comment, marked as answer]
This is by design, yes. Casks almost always do no more than what the upstream installer does. In some instances, they will write some symlinks into Homebrew's prefix for convenience, but I don't think any cask writes into `$HOME` without the upstream installer doing it.

Note that the `ncurses` formula does provide an `alacritty` terminfo entry.

**Reply by laggardkernel** [reply]
> Note that the `ncurses` formula does provide an `alacritty` terminfo entry.

`ncurses` is not linked, which means terminfo within is never searched by other application. In fact, you should never use terminfo from brew installed ncurses. It's incompatible with the old version in your system.

**Reply by carlocab** [reply]
I wouldn't say never. I use brewed `ncurses`' `terminfo` entry for `tmux-256color`, and aside from occasional hiccups that are easily avoided, it works very well. I did not recompile `tmux-256color` for old system `ncurses`.

Never using brewed `terminfo` probably also completely removes any benefits from a formula depending on the newer brewed `ncurses` instead of the old system one.

**Comment by laggardkernel** [comment]
@sluukkonen You can add the `TERMINFO_DIRS` to tell apps where to search the terminfo.

```bash
# WARNING: Don't ever set TERMINFO. Reading from `man terminfo`, when `TERMINFO` is set.
# all the following places are not searched anymore. Yes, `TERMINFO` shadows others.
# Use `TERMINFO_DIRS` instead.
# - $HOME/.terminfo
# - $TERMINFO_DIRS
# - /usr/share/terminfo

if [[ $SHLVL -eq 1 ]] && [[ $OSTYPE == darwin* ]]; then
  export TERMINFO_DIRS="/usr/share/terminfo"
  if [[ -d /Applications/Alacritty.app/Contents/Resources ]]; then
    export TERMINFO_DIRS="/Applications/Alacritty.app/Contents/Resources:${TERMINFO_DIRS}"
  fi
  if [[ -n $ALACRITTY_LOG ]]; then # shell start from Alacritty
    [[ -z $TMUX ]] && export TERM="alacritty"
    export LC_TERMINAL=Alacritty # not required, just my personal setting
  fi
fi
```

**Comment by ghost** [comment]
What I'm wondering is why aren't all "cursed" Homebrew packages curs^H^H^H^Hlinked with the Homebrew curses. The kind of maddeningly confusing issue reported [here](https://github.com/tmux/tmux/issues/2262) would occur less often, if they were. At least speaking for myself, the majority of programs I interact with on the command line come from Homebrew.
